File: app/src/main/java/com/lucidsoftworksllc/sabotcommunity/UsersApiClient.java
package com.lucidsoftworksllc.sabotcommunity;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class UsersApiClient {

    public static Retrofit retrofit;

    public static Retrofit getApiClient(){
        if (retrofit==null){
            retrofit = new Retrofit.Builder()
                    .baseUrl(Constants.ROOT_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
}

File: app/src/main/java/com/lucidsoftworksllc/sabotcommunity/UsersApiClient.kt
package com.lucidsoftworksllc.sabotcommunity

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object UsersApiClient {
    private var retrofit: Retrofit? = null
    val apiClient: Retrofit?
        get() {
            if (retrofit == null) {
                retrofit = Retrofit.Builder()
                        .baseUrl(Constants.ROOT_URL)
                        .addConverterFactory(GsonConverterFactory.create())
                        .build()
            }
            return retrofit
        }
}